pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -58,17 +58,13 @@ func (p *Pipeline) Validate(ctx context.Context) *apis.FieldError {
 	// Validate that all params and workspaces it uses are declared.
 	errs = errs.Also(p.Spec.validatePipelineParameterUsage(ctx).ViaField("spec"))
 	errs = errs.Also(p.Spec.validatePipelineWorkspacesUsage().ViaField("spec"))
-	// Validate beta fields when a Pipeline is defined, but not as part of validating a Pipeline spec.
-	// This prevents validation from failing when a Pipeline is converted to a different API version.
-	// See https://github.com/tektoncd/pipeline/issues/6616 for more information.
-	// TODO(#6592): Decouple API versioning from feature versioning
-	errs = errs.Also(p.Spec.ValidateBetaFields(ctx))
 	return errs
 }
 
 // Validate checks that taskNames in the Pipeline are valid and that the graph
 // of Tasks expressed in the Pipeline makes sense.
 func (ps *PipelineSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
+	errs = errs.Also(ps.ValidateBetaFields(ctx))
```

Comment:
It sounds good - any special reason for that? 
We might have beta fields in the status as well, but those are produced by the controller, so I guess it's fine to skip validation.

Reply:
Thanks for question Andrea. Related with the comment at https://github.com/tektoncd/pipeline/pull/6941#discussion_r1364619411 (many thanks to the suggestion from Priti), I think this was moved to the front for handling validations of easier logics first. And yes I do agree that validation for the PipelineSpec shall be sufficient for the webhook at the moment:)